File: frontend/src/app/services/_entity.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { UserModel } from './../models';
import 'rxjs/add/operator/do';
import { EntityModel } from '../models/_entity.model';

@Injectable()
export class EntityService<T extends EntityModel> {
  protected baseUrl = '/api/admin/*/';

  constructor(protected http: HttpClient) { }

  getAll() {
    return this.http.get<T[]>(this.baseUrl);
  }

  getById(id: number) {
    return this.http.get<T>(this.baseUrl + id);
  }

  create(entity: T) {
    return this.http.post(this.baseUrl, entity);
  }

  update(entity: T) {
    return this.http.put(this.baseUrl + entity.id, entity);
  }

  delete(id: number) {
    return this.http.delete(this.baseUrl + id);
  }
}

File: backend/src/main/java/com/cp/jworld/service/EntityService.java
package com.cp.jworld.service;

import com.cp.jworld.model.BaseEntity;

import java.util.List;

public interface EntityService<E extends BaseEntity> {
    List<E> getAll();

    E getById(Integer id);

    void insert(E entity);

    void update(E entity);

    void deleteById(Integer id);
}